File: pmd/java/listas/listaex02/ex1/Assistente.java
package listas.listaex02.ex1;

public class Assistente extends Funcionario {

  protected int matricula;

  public Assistente() {
    super();
    this.matricula = 0;
  }

  public Assistente(String nome, String email, int matricula) {
    super(nome, email);
    this.matricula = matricula;
  }

  public int getMatricula() {
    return matricula;
  }

  public void setMatricula(int matricula) {
    this.matricula = matricula;
  }

  public void exibeDados() {
    super.exibeDados();
    System.out.printf("\nMatricula: %d", matricula);
  }
}

File: pmd/python/listas/listaex02/ex1/assistente.py
from funcionario import Funcionario


class Assistente (Funcionario):

    def __init__(self, nome='', email='', matricula=0):
        super().__init__(nome, email)
        self.matricula = matricula

    def exibeDados(self):
        super().exibeDados()
        print('Matricula: %d' % self.matricula)


class AssistenteTecnico (Assistente):

    def __init__(self, nome='', email='', matricula=0, bonus=0.0):
        super().__init__(nome, email, matricula)
        self.bonus = bonus

    def exibeDados(self):
        super().exibeDados()
        print('Bonus: %.2f' % self.bonus)


class AssistenteAdministrativo (Assistente):

    def __init__(self, nome='', email='', matricula=0, turno='dia'):
        super().__init__(nome, email, matricula)
        self.turno = turno

    def exibeDados(self):
        super().exibeDados()
        print('Turno: %s' % self.turno)
